app/jobs/submission_invoice_creation_job.rb

```diff
@@ -1,5 +1,8 @@
 class SubmissionInvoiceCreationJob < ApplicationJob
   def perform(submission)
-    Workday::SubmitCustomerInvoiceRequest.new(submission).perform
+    return if submission.invoice.present?
+
+    workday_reference = Workday::SubmitCustomerInvoiceRequest.new(submission).perform
+    submission.invoice = SubmissionInvoice.new(workday_reference: workday_reference)
```

Comment:
Will this actually persist the invoice? I think perhaps this should be `submission.invoice.create(...)` or even `create!` so we know if something has gone wrong and the invoice doesn't get saved.

Reply:
**edit:**  I mean `submission.create_invoice!`

Reply:
I tried that first too! But has_one doesn't support that kind of join creation. This way does save it, as shown in the specs: https://github.com/dxw/DataSubmissionServiceAPI/pull/220/files#diff-52631cf0fbe8c36c509123d140e1c84aR21

But given that it caused confusion, I've tried another way, which is hopefully clearer, although does require a reload in the specs.